kube-derive/src/custom_resource.rs

```diff
@@ -447,7 +447,9 @@ pub(crate) fn derive(input: proc_macro2::TokenStream) -> proc_macro2::TokenStrea
 
             fn crd() -> #apiext::CustomResourceDefinition {
                 let columns : Vec<#apiext::CustomResourceColumnDefinition> = #serde_json::from_str(#printers).expect("valid printer column json");
-                let fields : Vec<#apiext::SelectableField> = #serde_json::from_str(#fields).expect("valid selectableField column json");
+                #k8s_openapi::k8s_if_ge_1_30! {
+                    let fields : Vec<#apiext::SelectableField> = #serde_json::from_str(#fields).expect("valid selectableField column json");
+                }
```

Comment:
yeah, i tried that also, but i couldn't get it compile because the macro seems to only work on blocks

Reply:
although it will fail against earliest:

```
$ cargo run --example crd_derive # with ealiest hacked into examples cargo.toml
error[E0425]: cannot find value `fields` in this scope
  --> examples/crd_derive.rs:12:10
   |
12 | #[derive(CustomResource, Serialize, Deserialize, Default, Debug, PartialEq, Eq, Clone, JsonSchema)]
   |          ^^^^^^^^^^^^^^ not found in this scope
```

Reply:
but, it is fine provided you don't use `selectable` against earliest!

Reply:
Yeah, this part is on the user and will not compile (although might need a better error message to understand the problem).

Reply:
I have an improvement for this. I'll raise a follow up PR with some error handling in kube-derive if you are happy with merging this.

Reply:
SGTM!